nats/js/client.py

```diff
@@ -243,12 +242,6 @@ async def cb(msg):
                 config = api.ConsumerConfig(
                     ack_policy=api.AckPolicy.EXPLICIT,
                 )
-            elif isinstance(config, dict):
-                config = api.ConsumerConfig.loads(**config)
-            elif not isinstance(config, api.ConsumerConfig):
-                raise ValueError("nats: invalid ConsumerConfig")
-            assert config is not None
```

Comment:
I decided to drop type checks there. If the type annotation says that `config` is a `ConsumerConfig`, it is the responsibility of the user (with the help of mypy) to ensure that's true.

Reply:
I liked being able to pass a dict for the ConsumerConfig instead, but I agree and instead need to add more options like in the Go client in case want to customize without using the type.

Reply:
Well, we can always support both and annotate it as `Union[ConsumerConfig, Dict[str, Any]]`. I'd keep it simple, though. Accepting `ConsumerConfig` makes it type-safe, gives the user a good autocomplete, points to the documentation, etc. A bit more verbose but it is Python, not Perl, here we keep things verbose if it is better for the reader. "Explicit is better than implicit".

In many places, we still accept `config, **params`, though, so the shorter version is still here, just not for this method. But again, having `**params: Any` will allow passing any arguments without type errors from the type checker and failing at runtime, which is not cool. So, I think we should have `**params` only for the methods that don't accept any other arguments except `config`, so passing any keyword arguments is a deliberate choice rather than a typo.